src/main/java/wolox/training/models/Book.java

```diff
@@ -116,39 +117,39 @@ public String getIsbn() {
     }
 
     private void setGenre(String genre) {
-        this.genre = genre;
+        this.genre = checkNotNull(genre);
```

Comment:
After checking if a field is null, check if the field is empty.

An error message can be added to each check :

```
Preconditions.checkNotNull(genre, "The genre field cannot be null.");
Preconditions.checkArgument(! genre.isEmpty(), "The genre field cannot be empty.");
this.username = username;
```

I recommend using a class to store error messages as constants.